x-pack/qa/repository-old-versions-7x/src/javaRestTest/java/org/elasticsearch/oldrepos7x/OldMappingsIT.java

```diff
@@ -0,0 +1,229 @@
+import org.elasticsearch.test.cluster.ElasticsearchCluster;
+import org.elasticsearch.test.cluster.local.distribution.DistributionType;
+import org.elasticsearch.test.cluster.util.Version;
+import org.elasticsearch.test.rest.ESRestTestCase;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentFactory;
+import org.elasticsearch.xcontent.XContentType;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TemporaryFolder;
+import org.junit.rules.TestRule;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class OldMappingsIT extends ESRestTestCase {
+
+    private static final List<String> indices = Arrays.asList("filebeat", "custom", "nested");
+
+    public static TemporaryFolder repoDirectory = new TemporaryFolder();
+
+    public static ElasticsearchCluster currentCluster = ElasticsearchCluster.local()
+        .distribution(DistributionType.DEFAULT)
+        .distribution(DistributionType.DEFAULT)
+        .setting("xpack.security.enabled", "false")
+        .setting("xpack.license.self_generated.type", "trial")
+        .setting("xpack.ml.enabled", "false")
+        .setting("path.repo", () -> repoDirectory.getRoot().getPath())
+        .build();
+
+    public static ElasticsearchCluster oldCluster = ElasticsearchCluster.local()
+        .version(Version.fromString("7.17.25"))
+        .distribution(DistributionType.DEFAULT)
+        .setting("xpack.security.enabled", "false")
+        .setting("xpack.license.self_generated.type", "trial")
+        .setting("xpack.ml.enabled", "false")
+        .setting("path.repo", () -> repoDirectory.getRoot().getPath())
+        .build();
+
+    @ClassRule
+    public static TestRule ruleChain = RuleChain.outerRule(repoDirectory).around(oldCluster).around(currentCluster);
+    private static boolean repoSetup = false;
+
+    @Override
+    protected String getTestRestCluster() {
+        return currentCluster.getHttpAddresses();
+    }
+
+    private Request createIndex(String indexName, String file) throws IOException {
+        Request createIndex = new Request("PUT", "/" + indexName);
+        int numberOfShards = randomIntBetween(1, 3);
+
+        XContentBuilder builder = XContentFactory.jsonBuilder()
+            .startObject()
+            .startObject("settings")
+            .field("index.number_of_shards", numberOfShards)
+            .field("index.number_of_replicas", 0)
+            .endObject()
+            .startObject("mappings");
+        builder.rawValue(OldMappingsIT.class.getResourceAsStream(file), XContentType.JSON);
+        builder.endObject().endObject();
+
+        createIndex.setJsonEntity(Strings.toString(builder));
+        return createIndex;
+    }
+
+    @Before
+    public void setupIndex() throws IOException {
+        if (repoSetup == false) {
```

Comment:
So far I failed to do the index setup in an @BeforeClass method so only have to create the snapshot once per test suite, so I worked around it with this flag here. I'm wondering though if there is a way to do the "old data setup" only once via some gradle mechanism so we only have to create the old repo data once and also are sure its there on disk before other tests with the "current" cluster run.